util/typeutil/stringInt.go

```diff
@@ -0,0 +1,32 @@
+package typeutil
+
+import (
+	"encoding/json"
+	"strconv"
+)
+
+type StringInt int
+
+func (st *StringInt) UnmarshalJSON(b []byte) error {
+	//convert the bytes into an interface as this will help us check the type of our value
+	var item interface{}
+	if err := json.Unmarshal(b, &item); err != nil {
+		return err
+	}
+
+	switch v := item.(type) {
+	case int:
+		*st = StringInt(v)
+	case float64:
+		*st = StringInt(int(v))
```

Comment:
Why do we need to support a float?

Reply:
good question. To Unmarshal Json in interface, Unmarshal stores JSON numbers in float64. Here is the official doc link - https://pkg.go.dev/encoding/json#Unmarshal